pyteal/compiler/flatten.py

```diff
@@ -123,7 +144,13 @@ def flattenSubroutines(
             if isinstance(stmt, TealLabel):
                 stmt.getLabelRef().addPrefix(labelPrefix)
 
-        combinedOps.append(TealLabel(None, LabelReference(label), comment))
+        dexpr: Expr | None = None
+        try:
+            dexpr = subroutine.declarations.get_declaration()
+        except TealInputError:
+            pass  # NOOP - just ignoring, as only used for source mapping
```

Comment:
Right. I had forgotten about this usage. So I should investigate whether the usage of frame-pointers is available at this juncture and instead call `subroutine.get_dcelrations_by_option(is_frame_ptr)`.

But then I'm concerned that this could trigger the same bug that we saw in #676 and that we talked about on Tuesday.

Reply:
keep me posted, I am curious of the implication of different compilation order.

Reply:
#682 gives an example of unanticipated side effects. I'm keeping this convo open as a visual reminder that we need to resolve this before merging. I think we should either modify the router to avoid this kind of side-effect, or ensure that the source mapper never creates a side effect.

Reply:
looking good, just one nit in https://github.com/algorand/pyteal/pull/650#discussion_r1128460365

Reply:
resolving.